R/glmnet.R

```diff
@@ -0,0 +1,146 @@
+                           multi = FALSE,
+                           ...) {
+
+  if (any(names(enquos(...)) == "newdata")) {
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+  }
+
+  # See discussion in https://github.com/tidymodels/parsnip/issues/195
+  if (is.null(penalty) & !is.null(object$spec$args$penalty)) {
+    penalty <- object$spec$args$penalty
+  }
+
+  object$spec$args$penalty <- .check_glmnet_penalty_predict(penalty, object, multi)
+
+  object$spec <- eval_args(object$spec)
+  predict.model_fit(object, new_data = new_data, type = type, opts = opts, ...)
+}
+
+predict_numeric_glmnet <- function(object, new_data, ...) {
+  if (any(names(enquos(...)) == "newdata"))
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+  object$spec <- eval_args(object$spec)
+  predict_numeric.model_fit(object, new_data = new_data, ...)
+}
+
+predict_class_glmnet <- function(object, new_data, ...) {
+  if (any(names(enquos(...)) == "newdata"))
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+  object$spec <- eval_args(object$spec)
+  predict_class.model_fit(object, new_data = new_data, ...)
+}
+
+predict_classprob_glmnet <- function(object, new_data, ...) {
+  if (any(names(enquos(...)) == "newdata"))
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+  object$spec <- eval_args(object$spec)
+  predict_classprob.model_fit(object, new_data = new_data, ...)
+}
+
+predict_raw_glmnet <- function(object, new_data, opts = list(), ...)  {
+  if (any(names(enquos(...)) == "newdata"))
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+  object$spec <- eval_args(object$spec)
+
+  opts$s <- object$spec$args$penalty
+
+  predict_raw.model_fit(object, new_data = new_data, opts = opts, ...)
+}
+
+
+
+
+multi_predict_glmnet <- function(object,
+                                 new_data,
+                                 type = NULL,
+                                 penalty = NULL,
+                                 ...) {
+
+  if (any(names(enquos(...)) == "newdata")) {
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+  }
+
+  if (object$spec$mode == "classification") {
+    if (is_quosure(penalty)) {
+      penalty <- eval_tidy(penalty)
+    }
+  }
+
+  dots <- list(...)
+
+  object$spec <- eval_args(object$spec)
+
+  if (is.null(penalty)) {
+    # See discussion in https://github.com/tidymodels/parsnip/issues/195
+    if (!is.null(object$spec$args$penalty)) {
+      penalty <- object$spec$args$penalty
+    } else {
+      penalty <- object$fit$lambda
+    }
+  }
+
+  if (object$spec$mode == "classification") {
+    if (is.null(type))
+      type <- "class"
+    if (!(type %in% c("class", "prob", "link", "raw"))) {
+      rlang::abort("`type` should be either 'class', 'link', 'raw', or 'prob'.")
+    }
+    if (type == "prob")
+      dots$type <- "response"
+    else
+      dots$type <- type
+  }
```

Comment:
I find this logic quite hard to read at first glance. I think it is because we are have a handful of logic back to back with no `{}`. I like this suggestion, but I also wouldn't hate a newline between each logic block

```suggestion
  if (object$spec$mode == "classification") {
    if (is.null(type)) {
      type <- "class"
    }
    if (!(type %in% c("class", "prob", "link", "raw"))) {
      rlang::abort("`type` should be either 'class', 'link', 'raw', or 'prob'.")
    }
    if (type == "prob") {
      dots$type <- "response"
    } else {
      dots$type <- type
    }
  }
```